lib/ansible/modules/service.py

```diff
@@ -1031,12 +1045,18 @@ def service_enable(self):
             if os.path.isfile(rcfile):
                 self.rcconf_file = rcfile
 
-        rc, stdout, stderr = self.execute_command("%s %s %s %s" % (self.svc_cmd, self.arguments, self.name, 'rcvar'))
+        cmd = f"{self.svc_cmd} "
+        # Jail name and verbose takes precedence over other arguments
+        if self.arguments.startswith(('-j', '-v')):
+            cmd += f"{self.arguments} {self.name} rcvar"
+        else:
+            cmd += f"{self.name} rcvar {self.arguments}"
+
+        rc, stdout, stderr = self.execute_command(cmd)
         try:
             rcvars = shlex.split(stdout, comments=True)
         except Exception:
-            # TODO: add a warning to the output with the failure
-            pass
+            self.module.fail_json(msg="failed to parse rcvar status", stdout=stdout, stderr=stderr)
 
         if not rcvars:
             self.module.fail_json(msg="unable to determine rcvar", stdout=stdout, stderr=stderr)
```

Comment:
I'm not sure the exception handlers here are useful. The existing error handler hasn't worked for ~10 years.

Unless we can demonstrate a failure on the split with integration tests, I'm inclined to just remove the exception handlers and let the default exception handler take care of things.

The original comment when the try/except was added was:

> avoid shlex exceptions in rc parsing (happens with multiline entries)

That might give you something to go on if you want to try to trigger a failure there... but if it's not easy, I'd just drop the exception handlers.

Reply:
I dropped exception all together since DT will handle it.